test/ArborX_EnableViewComparison.hpp

```diff
@@ -17,8 +17,88 @@
 #include <Kokkos_Core.hpp>
 
 #include "BoostTest_CUDA_clang_workarounds.hpp"
+#include <boost/test/unit_test.hpp>
 #include <boost/test/utils/is_forward_iterable.hpp>
 
+template <typename U, typename V>
+using CommonValueType =
+    typename boost::common_type<typename U::non_const_value_type,
+                                typename V::non_const_value_type>::type;
+
+template <typename U, typename V>
+void arborxViewCheck(U const &u, V const &v, std::string const &u_name,
+                     std::string const &v_name, CommonValueType<U, V> tol = 0)
+{
+  static constexpr int rank = U::rank;
+
+  bool same_dim_size = true;
+  for (int i = 0; i < rank; i++)
+  {
+    int ui = u.extent(i), vi = v.extent(i);
+    BOOST_TEST(ui == vi, "check " << u_name << " == " << v_name
+                                  << " failed at dimension " << i << " size ["
+                                  << ui << " != " << vi << "]");
+    same_dim_size = (ui == vi) && same_dim_size;
+  }
+
+  if (!same_dim_size)
+    return;
```

Comment:
I think that  you don't need this. `BOOST_TEST_REQUIRE` does what you want (to check).

Reply:
Not really, as this will abort the current test and not check the other matrices of the same test.